x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/Sqrt.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.function.scalar.math;
+
+import org.elasticsearch.compute.ann.Evaluator;
+import org.elasticsearch.compute.operator.EvalOperator;
+import org.elasticsearch.xpack.esql.expression.function.scalar.UnaryScalarFunction;
+import org.elasticsearch.xpack.esql.planner.Mappable;
+import org.elasticsearch.xpack.ql.expression.Expression;
+import org.elasticsearch.xpack.ql.tree.NodeInfo;
+import org.elasticsearch.xpack.ql.tree.Source;
+import org.elasticsearch.xpack.ql.type.DataType;
+import org.elasticsearch.xpack.ql.type.DataTypes;
+
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static org.elasticsearch.xpack.ql.expression.TypeResolutions.ParamOrdinal.DEFAULT;
+import static org.elasticsearch.xpack.ql.expression.TypeResolutions.isNumeric;
+
+public class Sqrt extends UnaryScalarFunction implements Mappable {
+    public Sqrt(Source source, Expression field) {
+        super(source, field);
+    }
+
+    @Override
+    public Supplier<EvalOperator.ExpressionEvaluator> toEvaluator(
+        Function<Expression, Supplier<EvalOperator.ExpressionEvaluator>> toEvaluator
+    ) {
+        Supplier<EvalOperator.ExpressionEvaluator> field = toEvaluator.apply(field());
+        var fieldType = field().dataType();
+        var eval = field.get();
+
+        if (fieldType == DataTypes.DOUBLE) {
+            return () -> new SqrtDoubleEvaluator(eval);
+        }
+        if (fieldType == DataTypes.INTEGER) {
+            return () -> new SqrtIntEvaluator(eval);
+        }
+        if (fieldType == DataTypes.LONG) {
+            return () -> new SqrtLongEvaluator(eval);
+        }
+
+        throw new UnsupportedOperationException("Unsupported type " + fieldType);
```

Comment:
The UNSIGNED_LONG support would need to be added too. OK if added in a subsequent PR.

Reply:
What's the suggested approach for handling it? Should it be handled as a long if less than Long. MAX_VALUE, otherwise as a double?

Since the rest of scalar functions don't handle it either, they can all be updated in a separate PR?

Reply:
I'd delay it, yeah. @bpintea maybe we need a meta issue to track missing functions for unsigned long. I imagine we have a few. And they are always kind of tricky to write.

Reply:
> the rest of scalar functions don't handle it either

@kkrik-es which functions do you refer to? I was hoping I covered them all, but that might not be the case and I'll open an issue for those.

> What's the suggested approach for handling it?

I think it'll eventually be fed to Math.sqrt() as a double, so converting to double should be fine imo.

Reply:
Right, that makes sense.

Wrt other functions, I see [Log10](x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/Log10.java) that I used as a template.

Reply:
@kkrik-es nice, yes, that's a bug. 
I did a quick pass and I hope I didn't miss any other numeric function not supporting UL right now.
If you want to have a look at both functions (i.e. this one and log10, which I think would be pretty similar) feel free to, otherwise I can have a look into log10 as well.

Reply:
I'm happy to take car of these two. I'll probably wait for merging to branch to main (if it hasn't happened) to avoid duplicate PRs.